test/cases/api.sh

```diff
@@ -531,6 +531,16 @@ esac
 # the server's response in case of an error.
 #
 
+function collectMetrics(){
+    METRICS_OUTPUT=$(curl \
+                          --cacert /etc/osbuild-composer/ca-crt.pem \
+                          --key /etc/osbuild-composer/client-key.pem \
+                          --cert /etc/osbuild-composer/client-crt.pem \
+                          https://localhost/metrics)
+
+    "$1"="$(echo "$METRICS_OUTPUT" | grep "^total_compose_requests" | cut -f2 -d' ')"
```

Comment:
Bash functions are a little quirky, they return whatever gets written to stdout in that function. It is possible to do what you're doing here using `declare -n out="$1"` and then `out="whatever value"` but I'm not super familiar with `declare`.

```
function barme () {
    declare -n out=$1
    out="bar"
}

FOO="foo"
declare -n FOO
barme FOO
echo $FOO # == bar
``` 

But this is difficult to read I think. Just using stdout will make a lot more sense to most people.

```
function collectMetrics() {
...
echo "$METRICS_OUTPUT" |  ... |cut -f2 -d' '
}
COMPOSES="$(collectMetrics)"
# or just
test "$(collectMetrics)" = "1"
```